src/app/qgshandlebadlayers.cpp

```diff
@@ -509,3 +495,164 @@ int QgsHandleBadLayers::layerCount()
 {
   return mLayerList->rowCount();
 }
+
+QString QgsHandleBadLayers::findFile( const QString &fileName, const QString &basePath, int maxDepth, int driveMargin )
```

Comment:
This function is quite general (and potentially useful outside of bad layer handler). Could you move it to QgsFileUtils?

Reply:
Absolutely, once this is fine I'll migrate.